Application/EdFi.Ods.AdminApi.AdminConsole/Infrastructure/Services/Instances/Validator/InstanceValidator.cs

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using EdFi.Ods.AdminApi.AdminConsole.Infrastructure.DataAccess.Models;
+using EdFi.Ods.AdminApi.AdminConsole.Infrastructure.Repositories;
+using EdFi.Ods.AdminApi.AdminConsole.Infrastructure.Services.Instances.Models;
+using FluentValidation;
+using Microsoft.EntityFrameworkCore;
+
+namespace EdFi.Ods.AdminApi.AdminConsole.Infrastructure.Services.Instances.Validator;
+
+public class InstanceValidator : AbstractValidator<IInstanceRequestModel>
+{
+    private readonly IQueriesRepository<Instance> _instanceQuery;
+    public InstanceValidator(IQueriesRepository<Instance> instanceQuery)
+    {
+        _instanceQuery = instanceQuery;
+
+        RuleFor(x => x.Name)
+            .NotEmpty()
+            .MaximumLength(100)
+            .MustAsync((instance, instanceName, cancellationToken) => BeUniqueForTenant(instance, instanceName, cancellationToken))
+            .WithMessage("The name must be unique for the tenant.");
+
+        RuleFor(x => x.InstanceType)
+            .NotEmpty()
+            .MaximumLength(100);
+
+        RuleFor(x => x.OdsInstanceContexts)
+            .Must(odsInstanceContexts => odsInstanceContexts == null || odsInstanceContexts.GroupBy(c => c.ContextKey).All(g => g.Count() == 1))
+            .WithMessage("ContextKey values must be unique within the list.");
+
+        RuleForEach(x => x.OdsInstanceContexts)
+            .SetValidator(new OdsInstanceContextValidator());
+
+        RuleForEach(x => x.OdsInstanceDerivatives)
+            .SetValidator(new OdsInstanceDerivativeValidator());
+    }
+
+    private async Task<bool> BeUniqueForTenant(IInstanceRequestModel instance, string instanceName, CancellationToken cancellationToken)
+    {
+        var tenantId = instance.TenantId;
+        var existingInstance = await _instanceQuery.Query()
+            .FirstOrDefaultAsync(i => i.InstanceName == instanceName && i.TenantId == tenantId, cancellationToken);
+        return existingInstance == null || existingInstance.Id == instance.OdsInstanceId;
```

Comment:
Aren't we mixing to different things here? `existingInstance.Id` is the admin console instance unique identifier and `instance.OdsInstanceId` is the reference to the `dbo.OdsInstances.OdsInstanceId`. I think.